src/app/shared/utils/utils.ts

```diff
@@ -324,9 +324,7 @@ export async function asyncEvery(arr: any[], predicate: Function) {
   return true;
 }
 
-/** A recursive version of Partial, making all properties, included nested ones, optional.
- * Copied from https://stackoverflow.com/a/47914631
- */
-export type RecursivePartial<T> = {
-  [P in keyof T]?: RecursivePartial<T[P]>;
-};
+/** Extract template name from a full router url */
+export function getTemplateNameFromUrl(urlString: string) {
+  return urlString.split(/[?#]/)[0].replace(/^(\/template\/)/, "");
+}
```

Comment:
nit(non-blocking)
I'm also wondering why we no longer need the RecursivePartial type - seemed like a nice utility to have. Has it been moved or just no longer required?